litestar/_asgi/routing_trie/mapping.py

```diff
@@ -182,38 +182,28 @@ def build_route_middleware_stack(
     Returns:
         An ASGIApp that is composed of a "stack" of middlewares.
     """
-    from litestar.middleware.allowed_hosts import AllowedHostsMiddleware
-    from litestar.middleware.compression import CompressionMiddleware
-    from litestar.middleware.csrf import CSRFMiddleware
-    from litestar.middleware.response_cache import ResponseCacheMiddleware
+
+    from litestar.contrib.opentelemetry import OpenTelemetryInstrumentationMiddleware
+    from litestar.middleware._internal.cors import CORSMiddleware
     from litestar.routes import HTTPRoute
 
     asgi_handler: ASGIApp = route.handle  # type: ignore[assignment]
     handler_middleware = route_handler.middleware
     has_cached_route = isinstance(route, HTTPRoute) and any(r.cache for r in route.route_handlers)
-    has_middleware = (
-        app.csrf_config or app.compression_config or has_cached_route or app.allowed_hosts or handler_middleware
-    )
-
-    if has_middleware:
-        # If there is an exception raised from the handler, the first ExceptionHandlerMiddleware that catches the
-        # exception will create the response and call send(). As middleware may wrap the send() callable, we need there
-        # to be an instance of ExceptionHandlerMiddleware in between the handler and the middleware so that any send
-        # wrappers instated by middleware are called. If there is no middleware, we can skip this step.
-        asgi_handler = wrap_in_exception_handler(app=asgi_handler)
-
-        if app.csrf_config:
-            asgi_handler = CSRFMiddleware(app=asgi_handler, config=app.csrf_config)
-
-        if app.compression_config:
-            asgi_handler = CompressionMiddleware(app=asgi_handler, config=app.compression_config)
-
-        if has_cached_route:
-            asgi_handler = ResponseCacheMiddleware(app=asgi_handler, config=app.response_cache_config)
-
-        if app.allowed_hosts:
-            asgi_handler = AllowedHostsMiddleware(app=asgi_handler, config=app.allowed_hosts)
```

Comment:
If we want the change to be backwards compatible, this would have to stay in until we deprecate the configs entirely. You think that's feasible, or too much work?

Reply:
I think it should be doable, Im gonna work on that asap

Reply:
ok this is interesing because putting this back made me found 3 tests that are still using `csrf_config` in test_openai/test_plugins

do we agree all tests should use the "new" way of setting this up and the "old" way is just tested by the deprecation messages ?

Reply:
done @provinzkraut ;)